pyocd/coresight/generic_mem_ap.py

```diff
@@ -0,0 +1,153 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+
+from .component import CoreSightCoreComponent
+from ..core.target import Target
+
+LOG = logging.getLogger(__name__)
+
+DEAD_VALUE = 0
+
+
+class GenericMemAP(Target, CoreSightCoreComponent):
+    def __init__(self, session, ap, memory_map=None, core_num=0, cmpid=None, address=None):
+        Target.__init__(self, session, memory_map)
+        CoreSightCoreComponent.__init__(self, ap, cmpid, address)
+        self.core_number = core_num
+        self.core_type = DEAD_VALUE
+        self._target_context = None
+        self.register_list = []
+
+    def add_child(self, cmp):
+        pass
+
+    @property
+    def supported_security_states(self):
+        return Target.SecurityState.NONSECURE,
+
+    def init(self):
+        pass
+
+    def disconnect(self, resume=True):
+        pass
+
+    def write_memory(self, addr, value, transfer_size=32):
+        self.ap.write_memory(addr, value, transfer_size)
+
+    def read_memory(self, addr, transfer_size=32, now=True):
+        return self.ap.read_memory(addr, transfer_size, True)
+
+    def read_memory_block8(self, addr, size):
+        return self.ap.read_memory_block8(addr, size)
+
+    def write_memory_block8(self, addr, data):
+        self.ap.write_memory_block8(addr, data)
+
+    def write_memory_block32(self, addr, data):
+        self.ap.write_memory_block32(addr, data)
+
+    def read_memory_block32(self, addr, size):
+        return self.ap.read_memory_block32(addr, size)
+
+    def halt(self):
+        pass
+
+    def step(self, disable_interrupts=True, start=0, end=0):
+        pass
+
+    def reset(self, reset_type=None):
+        pass
+
+    def reset_and_halt(self, reset_type=None):
+        self.reset(reset_type)
+        pass
+
+    def get_state(self):
+        return Target.State.HALTED
+
+    def get_security_state(self):
+        return Target.SecurityState.NONSECURE
+
+    def is_running(self):
+        return self.get_state() == Target.State.RUNNING
+
+    def is_halted(self):
+        return self.get_state() == Target.State.HALTED
+
+    def resume(self):
+        pass
+
+    def find_breakpoint(self, addr):
+        return None
+
+    def read_core_register(self, reg):
+        return DEAD_VALUE
+
+    def read_core_register_raw(self, reg):
+        return DEAD_VALUE
+
+    def read_core_registers_raw(self, reg_list):
```

Comment:
Just fyi on the r/w register methods… I'm working in the background on improving these methods so they can raise exceptions when reading or writing is invalid, and the rest of the code can handle that.